lib/sentry/logger_backend.ex

```diff
@@ -128,27 +97,54 @@ defmodule Sentry.LoggerBackend do
     :ok
   end
 
-  defp init(config, %__MODULE__{} = state) do
-    level = Keyword.get(config, :level, state.level)
+  defp log(level, msg, meta, state) do
+    sentry = meta[:sentry] || get_sentry_from_callers(meta[:callers] || [])
 
-    include_logger_metadata =
-      Keyword.get(config, :include_logger_metadata, state.include_logger_metadata)
+    opts =
+      [
+        event_source: :logger,
+        extra: %{logger_metadata: logger_metadata(meta, state), logger_level: level},
+        result: :none
+      ] ++ Map.to_list(sentry)
+
+    case meta[:crash_reason] do
+      {exception, stacktrace} when is_list(stacktrace) ->
+        Sentry.capture_exception(exception, [stacktrace: stacktrace] ++ opts)
+
+      reason when is_atom(reason) and not is_nil(reason) ->
+        Sentry.capture_exception(reason, opts)
```

Comment:
You can try this instead:

```elixir
defmodule Sentry.CrashError do
  defstruct [:message]

  def exception({:nocatch, reason}) do
    %Sentry.CrashError{message: Exception.format_banner(:throw, reason, [])}
  end

  def exception(reason) do
    %Sentry.CrashError{message: Exception.format_banner(:exit, reason, [])}
  end
end
```

and then here you do:

```suggestion
      {%_{__exception__: true} = exception, stacktrace} when is_list(stacktrace) ->
        Sentry.capture_exception(exception, [stacktrace: stacktrace] ++ opts)

      {other, stacktrace} when is_list(stacktrace) ->
        Sentry.capture_exception(Sentry.CrashError.exception(other), [stacktrace: stacktrace] ++ opts)
```